test/unit/org/apache/cassandra/index/sai/cql/LuceneUpdateDeleteTest.java

```diff
@@ -25,10 +25,43 @@
 
 import static org.apache.cassandra.index.sai.cql.VectorTypeTest.assertContainsInt;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.Assert.assertEquals;
 
 public class LuceneUpdateDeleteTest extends SAITester
 {
+    @Test
+    public void updateAndDeleteWithAnalyzerRestrictionQueryShouldFail() throws Throwable
+    {
+        createTable("CREATE TABLE %s (id int PRIMARY KEY, val text)");
+
+        createIndex("CREATE CUSTOM INDEX ON %s(val) " +
+                    "USING 'org.apache.cassandra.index.sai.StorageAttachedIndex' " +
+                    "WITH OPTIONS = { 'index_analyzer': '[{\"tokenizer\": \"standard\"}, {\"filter\": \"lowercase\"}]' }");
+
+        waitForIndexQueryable();
+
+        execute("INSERT INTO %s (id, val) VALUES (0, 'a sad doG.')");
+
+        // Prove we can get the row back
+        assertEquals(1, execute("SELECT * FROM %s WHERE val : 'dog'").size());
+
+        // DELETE fails
+        assertThatThrownBy(() -> execute("DELETE FROM %s WHERE val : 'dog'"))
+        .isInstanceOf(RuntimeException.class)
```

Comment:
The exception should be an `InvalidRequestException`
Same thing for the 2 others test bellow.

Reply:
Thanks, looks like that was the exception being thrown already. Not sure where I pulled the `RuntimeException` from.